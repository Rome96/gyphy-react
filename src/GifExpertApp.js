import React, {useState} from 'react'
import GifGrid from './components/GifGrid';
import AddCategory from './components/AddCategory'

const GifExpertApp = () => {
  const [categories, setCategories] = useState(['naruto'])

  // const handleAdd = () => {
  //   setCategories([
  //     'Turiano',
  //     ...categories,
  //   ])
  // }

  return (
    <>
      <h2>Gif Expert App</h2>
      <AddCategory setCategories={setCategories}/>
      <hr />
      <ol>
        {categories.map((category, i) => (
          <GifGrid
            key={i}
            category={category}
          />
        ))}
      </ol>
    </>
  );
}

export default GifExpertApp
