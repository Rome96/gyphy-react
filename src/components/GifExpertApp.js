import React from 'react'
// import PropTypes from 'prop-types'

const GifExpertApp = props => {
  return (
    <>
      <h2>Gif Expert App</h2>
    </>
  )
}

GifExpertApp.propTypes = {

}

export default GifExpertApp
