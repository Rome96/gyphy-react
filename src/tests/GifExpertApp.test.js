import React from 'react'
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';
import '@testing-library/jest-dom'
describe('testing component <GifExpertApp/>', () => {

	test('return component <GifExpertApp/>', () => {
		const wrapper = shallow(<GifExpertApp/>)
		expect(wrapper).toMatchSnapshot()
	});
});