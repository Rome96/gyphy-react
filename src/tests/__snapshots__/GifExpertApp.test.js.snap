// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing component <GifExpertApp/> return component <GifExpertApp/> 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Goku"
      key="0"
    />
  </ol>
</Fragment>
`;
